midi/controller/UndoRedoStack.h

```diff
@@ -20,16 +22,19 @@ using MidiSequencer4Ptr = std::shared_ptr<MidiSequencer4>;
 class UndoRedoStack
 {
 public:
+    UndoRedoStack() {
+         moduleId = 0;
+    }
     // It's a bit of a hack to have a version that doesn't require a widget,
     // But since the widget param is rarely used... 
     // execute the command, make undo record
     void execute(MidiSequencerPtr, std::shared_ptr<SqCommand>);
     void execute(MidiSequencerPtr, SequencerWidget*, std::shared_ptr<SqCommand>);
     void execute4(MidiSequencer4Ptr, std::shared_ptr<Sq4Command>);
     void execute4(MidiSequencer4Ptr, Sequencer4Widget*, std::shared_ptr<Sq4Command>);
-    void setModuleId(int);
+    void setModuleId(int64_t, bool fromUI);
 private:
-    int moduleId=-1;
+    std::atomic<int64_t> moduleId;
```

Comment:
This is the main issue. in VCV 1.0, module Id's were 32 bits. In 2.0 they expanded to 64 bits. No, this is not in the porting guide.

You can see, too that I am using std::atomic here. That's because some other changes made it so multiple threads can update this ID at the same time.